### PR TITLE
[OTB] Fixes empty string parameters

### PR DESCRIPTION
**Pull request by troopa81**
Fixes OTB/QGIS [issue](https://gitlab.orfeo-toolbox.org/orfeotoolbox/otb/-/issues/2130) issue, consequence of #39939.

to sum up, we don't have to pass *None* parameters or empty string parameters to OTB, but we have to take care of the zero integer value (my previous fix). 

I added some tests for non regression.

### CODE REVIEW

**Comment by nyalldawson** [comment]
Can you fix the tour typo in the comment above this line (business)

File: python/plugins/processing/algs/otb/OtbAlgorithm.py [hunk 1]
```diff
@@ -204,8 +204,8 @@ def processAlgorithm(self, parameters, context, feedback):
         command = '"{}" {} {}'.format(app_launcher_path, self.name(), OtbUtils.appFolder())
         outputPixelType = None
         for k, v in parameters.items():
-            # if value is None for a parameter we don't have any businees with this key
-            if v is None:
+            # if value is None or en empty string for a parameter we don't have any businees with this key
```
Comment (nyalldawson) [inline python/plugins/processing/algs/otb/OtbAlgorithm.py:207]:
```suggestion
            # if value is None or en empty string for a parameter then we don't want to pass that value to OTB (see https://gitlab.orfeo-toolbox.org/orfeotoolbox/otb/-/issues/2130)
```

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
Manually backported as 6c2b26ecaea85ae59278bcc0188b78311db53ba0

**Comment by troopa81** [comment]
Thanks @nyalldawson

**Pull request merged**